Source/Core/Core/Lua/LuaFunctions/LuaEmuFunctions.cpp

```diff
@@ -0,0 +1,137 @@
+#include "LuaEmuFunctions.h"
+#include "../LuaVersionResolver.h"
+#include "Core/State.h"
+#include "Core/Core.h"
+#include "Core/Movie.h"
+#include "Core/PowerPC/PowerPC.h"
+#include <optional>
+
+namespace Lua
+{
+namespace LuaEmu
+{
+  bool waitingForSaveStateLoad = false;
+  bool waitingForSaveStateSave = false;
+  bool waitingToStartPlayingMovie = false;
+  bool waitingToSaveMovie = false;
+
+  std::string loadStateName;
+  std::string saveStateName;
+  std::string moviePathName;
+  std::string playMovieName;
+  std::optional<std::string> blankString;
+  std::string saveMovieName;
+
+  class emu
+{
+public:
+  inline emu() {}
+};
+
+emu* emuPointer = NULL;
+
+emu* GetEmuInstance()
+{
+  if (emuPointer == NULL)
+    emuPointer = new emu();
+  return emuPointer;
+}
+
+void InitLuaEmuFunctions(lua_State* luaState, const std::string& luaApiVersion)
+{
+  waitingForSaveStateLoad = false;
+  waitingForSaveStateSave = false;
+  waitingToStartPlayingMovie = false;
+  waitingToSaveMovie = false;
+  emu** emuPtrPtr = (emu**)lua_newuserdata(luaState, sizeof(emu*));
+  *emuPtrPtr = GetEmuInstance();
+  luaL_newmetatable(luaState, "LuaEmuMetaTable");
+  lua_pushvalue(luaState, -1);
+  lua_setfield(luaState, -2, "__index");
+
+  std::array luaEmuFunctionsWithVersionsAttached = {
+    luaL_Reg_With_Version({"frameAdvance", "1.0", emu_frameAdvance}),
+    luaL_Reg_With_Version({"loadState", "1.0", emu_loadState}),
+    luaL_Reg_With_Version({"saveState", "1.0", emu_saveState}),
+    luaL_Reg_With_Version({"playMovie", "1.0", emu_playMovie}),
+    luaL_Reg_With_Version({"saveMovie", "1.0", emu_saveMovie}),
+   };
+
+  std::unordered_map<std::string, std::string> deprecatedFunctionsMap = std::unordered_map<std::string, std::string>();
+  addLatestFunctionsForVersion(luaEmuFunctionsWithVersionsAttached, luaApiVersion, deprecatedFunctionsMap, luaState);
+  lua_setglobal(luaState, "emu");
+}
+
+void StatePauseFunction()
+{
+  Core::SetState(Core::State::Paused);
+}
+
+int emu_frameAdvance(lua_State* luaState)
+{
+  luaColonOperatorTypeCheck(luaState, "frameAdvance", "emu:frameAdvance()");
+  return lua_yield(luaState, 0);
+}
+
+void convertToUpperCase(std::string& inputString)
```

Comment:
We have a function for this in [StringUtil.h](https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/Common/StringUtil.h#L266)